src/packagers/JSConcatPackager.js

```diff
@@ -23,6 +24,31 @@ const helpers =
     .readFileSync(path.join(__dirname, '../builtins/helpers.js'), 'utf8')
     .trim() + '\n';
 
+async function hasSideEffects(asset, sideEffects) {
```

Comment:
nit: defaulting `sideEffects` using `{sideEffects} = asset.cacheData` could make the `if` conditions using it cleaner (`await hasSideEffects(asset)`)